tools/common/Application.cs

```diff
@@ -396,6 +396,10 @@ public static void TryDelete (string path)
 
 		public void InitializeCommon ()
 		{
+			if (LinkMode == LinkMode.None && SdkVersion < SdkVersions.GetVersion (Platform)) {
```

Comment:
For Xamarin.Mac using the dynamic registrar is a viable alternative; which means you should probably only do this check when using the static registrar (this should be reflected in the error documentation as well; maybe mentioning that the static registrar produces optimized code (which would explain why we enable it for release builds), but the downside is that it requires an up-to-date Xcode).